p2p/peer.py

```diff
@@ -128,13 +128,14 @@
          ) = await auth.handshake(remote, privkey, token)
     except (ConnectionRefusedError, OSError) as e:
         raise UnreachablePeer() from e
-    except MalformedMessage as e:
-        raise HandshakeFailure() from e
     peer = peer_class(
         remote=remote, privkey=privkey, reader=reader, writer=writer,
         aes_secret=aes_secret, mac_secret=mac_secret, egress_mac=egress_mac,
         ingress_mac=ingress_mac, headerdb=headerdb, network_id=network_id)
-    await peer.do_p2p_handshake()
+    try:
+        await peer.do_p2p_handshake()
+    except MalformedMessage as e:
```

Comment:
It'd be better to do that in `Peer.do_p2p_handshake()`, to avoid having to duplicate the try/except everywhere we call it

Reply:
Alright, mind taking another look at that last wip commit?